packages/venia-concept/webpack.config.js

```diff
@@ -105,30 +106,48 @@ module.exports = async function(passedEnv) {
             }
         }),
         plugins: [
-            new MagentoRootComponentsPlugin({ phase }),
-            new webpack.NoEmitOnErrorsPlugin(),
+            await makeMagentoRootComponentsPlugin({
+                rootComponentsDirs,
+                context: __dirname
+            }),
             new webpack.DefinePlugin({
-                'process.env.NODE_ENV': JSON.stringify(phase),
-                // Blank the service worker file name to stop the app from
-                // attempting to register a service worker in index.js.
-                // Only register a service worker when in production or in the
-                // special case of debugging the service worker itself.
-                'process.env.SERVICE_WORKER': JSON.stringify(
-                    phase === 'production' || enableServiceWorkerDebugging
-                        ? serviceWorkerFileName
-                        : false
-                )
+                'process.env': {
+                    NODE_ENV: JSON.stringify(mode),
+                    // Blank the service worker file name to stop the app from
+                    // attempting to register a service worker in index.js.
+                    // Only register a service worker when in production or in the
+                    // special case of debugging the service worker itself.
+                    SERVICE_WORKER: JSON.stringify(
+                        mode === 'production' || enableServiceWorkerDebugging
+                            ? serviceWorkerFileName
+                            : false
+                    )
+                }
             }),
             new ServiceWorkerPlugin({
-                env: Object.assign({}, validEnv, passedEnv),
+                env: { mode },
                 enableServiceWorkerDebugging,
                 serviceWorkerFileName,
                 paths: themePaths
             })
-        ]
+        ],
+        optimization: {
+            splitChunks: {
+                cacheGroups: {
+                    vendor: {
+                        test: new RegExp(
+                            `[\\\/]node_modules[\\\/](${libs.join('|')})[\\\/]`
+                        ),
+                        name: 'vendor',
+                        filename: 'js/vendor.js',
+                        chunks: 'all'
+                    }
+                }
+            }
+        }
     };
-    if (phase === 'development') {
-        config.devtool = 'eval-source-map';
+    if (mode === 'development') {
+        config.devtool = 'inline-source-map';
```

Comment:
Why? The original setting, `eval-source-map`, seems to be the same but strictly faster.

Reply:
Whoops, didn't mean to check this in.